lib/api/2.0/templates.js

```diff
@@ -26,23 +26,15 @@ var templatesMetaGetByName = controller(function(req) {
 
 // GET /templates/:name
 var templatesGetByName = controller(function(req, res) {
-    return lookups.ipAddressToMacAddress(res.locals.ipAddress)
-    .then(function(mac) {
-        return waterline.nodes.needByIdentifier(mac);
-    })
-    .then(function(node) {
-        return Promise.all([
-            workflowApiService.findActiveGraphForTarget(node.id),
-            node
-        ]);
-    })
-    .spread(function(workflow, node) {
+    var nodeId = req.swagger.query.nodeId;
```

Comment:
We have created story https://www.pivotaltracker.com/story/show/134938873 to follow the suggestions.
Thanks.